### PR TITLE
feat(clear-topic): add back button when qns are filtered by topic

### LINKED ISSUE #458: Allow for users to go back after a 'Topic' is selected
Currently after selecting a `Topic`, users have no way to undo that action and go back to `All Questions` page

Can we add a `Back to all questions`

### PR DESCRIPTION
**Pull request by LinHuiqing**
## Problem

Users are unable to clear topic after a topic was previously selected.

Closes #458

## Solution

Add a back to home button to the page that shows questions cleared by topic.

**Features**:

- render BackToHome on HomePage when qns filtered by tag

**Improvements**:

- refactor BackToHome to use chakra and take in param for main page to go to
- refactor NotFound, Post and SearchResults for back button

## Before & After Screenshots

**BEFORE**:

Mobile:

https://user-images.githubusercontent.com/37061143/135782270-ef8573d7-8dce-4a30-b8f3-363c2db11ae6.mov

Desktop:

https://user-images.githubusercontent.com/37061143/135782283-6848b1c9-5322-4f8f-9ce7-3c0069ba1479.mov

**AFTER**:

Mobile:

https://user-images.githubusercontent.com/37061143/135781640-67310650-7ab3-4c86-ae24-90d360f15179.mov

Desktop:

https://user-images.githubusercontent.com/37061143/135781645-cfc20a39-5f48-46b5-859d-6834d4772ab4.mov

## Tests

1. Select a topic either from the panel or from under a question. The user should see a back button at the top of the topic. This button should direct the user back to their main page.
2. After selecting a question, there should be a back button directing to the main page which the question is from.
3. At the search results page, there should be a back button directing to the main page which the search was made.
4. At a not found page, there should be a back button directing to the main AskGov page.

### CODE REVIEW

File: client/src/components/BackToHome/BackToHome.tsx [hunk 1]
```diff
@@ -0,0 +1,22 @@
...
+}: {
+  mainPageName: string | null
+}) => {
+  return (
+    <Link as={RouterLink} to={mainPageName ? `/agency/${mainPageName}` : '/'}>
+      <Flex direction="row">
+        <Center color="secondary.600">
+          <BiArrowBack style={{ marginRight: '14px' }} size="13.41px" />
+          <Text>
+            Back to {mainPageName ? mainPageName.toUpperCase() : ''} questions
```
Comment (LoneRifle) [inline client/src/components/BackToHome/BackToHome.tsx]:
```suggestion
            Back to {mainPageName?.toUpperCase() ?? ''} questions
```

**Review by LoneRifle — APPROVED** [review]
Lgtm otherwise

File: client/src/components/BackToHome/BackToHome.tsx (hunk 1 above)
Comment (LinHuiqing) [reply client/src/components/BackToHome/BackToHome.tsx]: Okay, addressed in amended commit https://github.com/opengovsg/askgovsg/commit/efb20fe576a54e77c1f5f7284b35bb3344fc2adb.

**Comment by khaleedah** [comment]
Can we standardize spacing between `<- Back to MOH Questions` and `Questions related to <topic>` to same as question detail page?

Might have further wording changes to `Questions related to <topic>` but I'll discuss this separately

**Pull request merged**